pkg/tools/gen/genkcl.go

```diff
@@ -0,0 +1,169 @@
+package gen
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"kusionstack.io/kclvm-go/pkg/logger"
+)
+
+// TODO: is need to parse from Comment. Now get kcl schema from Tag
+type GenKclOptions struct {
+	KclPackage       string
+	ParseFromComment bool
+	ParseFromTag     bool
+}
+
+type kclGenerator struct {
+	opts *GenKclOptions
+}
+
+// GenKcl translate go struct to kcl schema code.
+func GenKcl(w io.Writer, filename string, src interface{}, opts *GenKclOptions) error {
+	return newKclGenerator(opts).GenSchema(w, filename, src)
+}
+
+func newKclGenerator(opts *GenKclOptions) *kclGenerator {
+	if opts == nil {
+		opts = new(GenKclOptions)
+	}
+	return &kclGenerator{
+		opts: opts,
+	}
+}
+
+func (k *kclGenerator) GenSchema(w io.Writer, filename string, src interface{}) error {
+	fmt.Fprintln(w)
+	goStructs, err := ParseGoSourceCode(filename, src)
+	if err != nil {
+		return err
+	}
+	for _, goStruct := range goStructs {
+		fmt.Fprintf(w, "schema %s:\n", goStruct.Name)
+		if goStruct.StructComment != "" {
+			fmt.Fprintf(w, "\"\"\"%s\"\"\"\n", goStruct.StructComment)
+		}
+		for _, field := range goStruct.Fields {
+			kclFieldName, kclFieldType, err := k.parserGoStructFieldTag(field.FieldTag)
```

Comment:
If there is no tag string written here, can we directly use the name and type of the go field?

Reply:
Will implement it.

Reply:
implement generate kcl code from struct field info.